ui/src/app/shared/services/applications.service.ts

```diff
@@ -123,6 +124,10 @@ export class ApplicationsService {
         return this.userApplications;
     }
 
+    public getAvailableApplicationMetadata(environmentId: string, topicOwnerAppId: string, topicName: string): Observable<any> {
+        return this.http.get(`/api/applicationsmetadata/${environmentId}/${topicName}/?topicOwnerId=${topicOwnerAppId}`);
+    }
+
```

Comment:
As already commented in the backend, should be implemented by frontend logic. Also, please, in topics.service.ts, not in applications.service.ts, as this is topic specific.